src/NServiceBus.Transport.SQS/InputQueuePump.cs

```diff
@@ -391,29 +417,32 @@ async Task MovePoisonMessageToErrorQueue(Message message)
             }
             catch (Exception ex)
             {
-                Logger.Warn($"Error removing poison message from input queue {inputQueueUrl}. This may cause duplicate poison messages in the error queue for this endpoint.", ex);
+                _logger.Warn($"Error removing poison message from input queue {inputQueueUrl}. This may cause duplicate poison messages in the error queue for this endpoint.", ex);
             }
 
             // If there is a message body in S3, simply leave it there
         }
 
         List<Task> pumpTasks;
-        Func<ErrorContext, Task<ErrorHandleResult>> onError;
-        Func<MessageContext, Task> onMessage;
+        OnError onError;
+        OnMessage onMessage;
         SemaphoreSlim maxConcurrencySemaphore;
         string inputQueueUrl;
         string errorQueueUrl;
         int maxConcurrency;
-        TransportConfiguration configuration;
-        IAmazonS3 s3Client;
-        IAmazonSQS sqsClient;
-        QueueCache queueCache;
+
+        readonly ReceiveSettings settings;
+        readonly IAmazonSQS sqsClient;
+        readonly QueueCache queueCache;
+        readonly S3Settings s3Settings;
+        readonly Action<string, Exception> criticalErrorAction;
+        readonly string awsEndpointUrl;
+
         int numberOfMessagesToFetch;
         ReceiveMessageRequest receiveMessagesRequest;
-        CriticalError criticalError;
-        string awsEndpointUrl;
+        CancellationTokenSource tokenSource;
 
-        static readonly TransportTransaction transportTransaction = new TransportTransaction();
-        static ILog Logger = LogManager.GetLogger(typeof(MessagePump));
+        static readonly TransportTransaction TransportTransaction = new TransportTransaction();
```

Comment:
Ups I think we have a bug here that is already present in the released version. This should not be static! Smells like almost a critical honestly